wholegames-vcr.Rmd

```diff
@@ -0,0 +1,300 @@
+        Requests","status":"operational","created_at":"2020-09-02T15:39:06.329Z","updated_at":"2020-10-10T00:02:49.033Z","position":6,"description":"Requests
+        for Pull Requests on GitHub.com","showcase":false,"start_date":null,"group_id":null,"page_id":"kctbh9vrtdwd","group":false,"only_show_if_degraded":false},{"id":"br0l2tvcx85d","name":"GitHub
+        Actions","status":"operational","created_at":"2019-11-13T18:02:19.432Z","updated_at":"2020-10-13T20:23:36.040Z","position":7,"description":"Workflows,
+        Compute and Orchestration for GitHub Actions","showcase":false,"start_date":null,"group_id":null,"page_id":"kctbh9vrtdwd","group":false,"only_show_if_degraded":false},{"id":"st3j38cctv9l","name":"GitHub
+        Packages","status":"operational","created_at":"2019-11-13T18:02:40.064Z","updated_at":"2020-09-08T15:50:32.845Z","position":8,"description":"API
+        requests and webhook delivery for GitHub Packages","showcase":false,"start_date":null,"group_id":null,"page_id":"kctbh9vrtdwd","group":false,"only_show_if_degraded":false},{"id":"vg70hn9s2tyj","name":"GitHub
+        Pages","status":"operational","created_at":"2017-01-31T20:04:33.923Z","updated_at":"2020-10-10T00:02:38.220Z","position":9,"description":"Frontend
+        application and API servers for Pages builds","showcase":false,"start_date":null,"group_id":null,"page_id":"kctbh9vrtdwd","group":false,"only_show_if_degraded":false}]}'
+  recorded_at: 2020-10-15 11:59:23 GMT
+  recorded_with: vcr/0.5.4, webmockr/0.7.0
+```
+
+* all the times after that, unless we delete the mock file, vcr simply uses the mock files instead of actually calling the API.
+
+Let's tweak our other test, of `gh_organizations()`. 
+Here things get more exciting or complicated, as we also set out to adding a test of the error behavior.
+This inspired us to change error behavior a bit with a slightly more specific error message i.e. `httr::stop_for_status(response)` became `httr::stop_for_status(response, task = "get data from the API, oops")`. 
+
+The test file ` tests/testthat/test-organizations.R` is now:
+
+```r
+test_that("gh_organizations works", {
+  vcr::use_cassette("gh_organizations", {
+    orgs <- gh_organizations()
+  })
+  testthat::expect_type(orgs, "character")
+})
+
+test_that("gh_organizations errors when the API doesn't behave", {
+  webmockr::enable()
+  stub <- webmockr::stub_request("get", "https://api.github.com/organizations?since=1")
+  webmockr::to_return(stub, status = 502)
+  expect_error(gh_organizations(), "oops")
+  webmockr::disable()
+})
+
+```
+
+The first test is similar to what we did for `gh_api_status()`.
+In the second test there is more to unpack.
+
+* We enable the use of `{webmockr}` at the beginning with `webmockr::enable()`. Why webmockr? Because it's the package powering request stubbing in vcr so vcr and webmockr go well together.
```

Comment:
Why webmockr? The reason given after that doesn't seem quite right. Although true that webmockr is used in vcr, I would think the reason for using webmockr in this use case is because it can help mock a failure scenario.